app/gui2/src/stores/graph/graphDatabase.ts

```diff
@@ -298,6 +334,12 @@ export class GraphDb {
     return currentNodeIds
   }
 
+  private deleteNode(nodeId: ExprId) {
+    this.nodeIdToNode.delete(nodeId)
+    this.nodeRects.delete(nodeId)
+    this.vizRects.delete(nodeId)
```

Comment:
Cleanup of node/visualization rects should be done at the view level, using `onUnmounted` hooks (or something else with the same effect). Similarily to what is currently done in port widgets managing their `portId` to view registration using `addPortInstance` and `removePortInstance`.

Reply:
Done